packages/chain-adapters/src/types.ts

```diff
@@ -85,54 +85,30 @@ export type SubscribeTxsInput = {
   accountType?: UtxoAccountType
 }
 
-export type TxFee = {
-  assetId: string
-  value: string
-}
+export type TransferType = unchained.TransferType
+export type TradeType = unchained.TradeType
 
-export type Transaction<T extends ChainId> = {
-  address: string
-  blockHash?: string
-  blockHeight: number
-  blockTime: number
-  chain: T
-  chainId: string
-  confirmations: number
-  txid: string
-  fee?: TxFee
-  status: TxStatus
-  tradeDetails?: TradeDetails
+export type Transaction = Omit<unchained.StandardTx, 'transfers'> & {
   transfers: Array<TxTransfer>
-  data?: TxMetadata
+  data?: unchained.StandardTxMetadata | unchained.evm.TxMetadata | unchained.cosmos.TxMetadata
 }
 
-export type TradeDetails = {
-  dexName: string
-  memo?: string
-  type: TradeType
-}
-
-export interface TxMetadata {
-  method?: string
-  parser: string
-}
+export type TransactionMetadataStandard = unchained.StandardTxMetadata
+export type TransactionMetadataEvm = unchained.evm.ERC20TxMetadata
+export type TransactionMetadataCosmos = unchained.cosmos.TxMetadata
```

Comment:
would like these ecosystem specific types to live in their corresponding `evm/types.ts` or `cosmossdk/type.ts` files that can be used/exported as needed (can add an evm and cosmossdk export alias to nest these ecosystem types under to keep nice namespacing)

Reply:
not sure how well I described this so happy to talk through, might be easier haha

Reply:
Discussed this one out and decided to keep only one terminology i.e `TransactionMetadata` - using namespacing at import time: https://github.com/shapeshift/lib/pull/944/commits/e87cf4ca2f6ff35e11130641065a240e4e518a45 and https://github.com/shapeshift/lib/pull/944/commits/6c1d905b126eccca2791e5ada10a4462d7a86df1